### DISCUSSION TITLE
mobilityDB and PostgreSQL "make installcheck"

### DISCUSSION THREAD

**Discussion by c2main** [ULB-CoDE-WIT/MobilityDB#363] [General]
Hi,

I was playing with (excellent) mobilityDB, and looking at the current debian packaging when some question raised about the way tests are done.

I made a quick test to migrate what is included in `sql/general` and it looks good on a first look. I've noticed there are some plays with/without "fast test" (?), without MEOS maybe, and some test with condition execution but I didn't explore that much.

I wonder if there is a known blocker to use the PostgreSQL PGXS makefile for running `make installcheck` ?
Else will you be interested in using PGXS ? (for reference: <https://www.postgresql.org/docs/current/extend-pgxs.html> )

**Comment by estebanzimanyi** [comment]
Many thanks Cédric for your interest in MobilityDB!
* I didn't understand what do you mean with "fast test" :-(
* I don't know see any blocker to use PGXS but I am not an expert on this. Any help in this respect is more than welcome!

For your information, the relationship between MobilityDB and MEOS is explained in the following document
https://github.com/MobilityDB/MobilityDB/wiki/Building-MobilityDB-and-MEOS

**Reply by c2main** [reply]
Details are interesting indeed, explanations of source directories very welcome. Thank you for the pointer.

 I have completed the migration of the SQL regression tests using PostgreSQL makefiles (PGXS), looks good so far. I'm still reading to evaluate how complex it'll be to use PGXS completely... 

About the "fast test": it's just that by looking at what exist in the build dir I noticed make "targets" named "test/fast" (and I don't know Cmake so maybe it's just some usual target it produces).

**Reply by c2main** [reply]
results: I am more looking the other way around now ... the regression tests are working as expected with "pg_regress" so it looks much better to update CMake to use it...

**Comment by estebanzimanyi** [comment]
How can help you on this regard ? I am not sure that understands all the details ....

**Reply by c2main** [reply]
well, your feedback is helpful. I was diving the code and was wondering what lead MobilityDB to use it's own test framework.
It's pretty clear now, hopefully I'll submit some patches to just simplify the current code.

**Comment by estebanzimanyi** [comment]
Many many thanks for your commitment Cédric !

**Comment by estebanzimanyi** [comment]
MobilityDB's regression tests are run through its own CMake and CTest harness, which is the supported path today; there is no active plan to migrate the test invocation to the PGXS make installcheck flow. Closing this dormant question as outdated; a focused issue or PR is welcome if PGXS support is something you would like to contribute.
